utils/src/test/scala/com/salesforce/op/utils/stats/StreamingHistogramTest.scala

```diff
@@ -0,0 +1,250 @@
+    cdfTest("Beta(1, 5)", beta2, histogramSampleSize, 200, mcSampleSize, 50)
+    cdfTest("Beta(1, 5)", beta2, histogramSampleSize, 400, mcSampleSize, 50)
+    cdfTest("0.5 * Beta(5, 1) + 0.5 * Beta(1, 5)", mixture, histogramSampleSize, 100, mcSampleSize, 50)
+    cdfTest("0.5 * Beta(5, 1) + 0.5 * Beta(1, 5)", mixture, histogramSampleSize, 200, mcSampleSize, 50)
+    cdfTest("0.5 * Beta(5, 1) + 0.5 * Beta(1, 5)", mixture, histogramSampleSize, 400, mcSampleSize, 50)
+  }
+
+
+  private def cdfTest(
+    distributionName: String,
+    dist: TestDistribution,
+    histogramSampleSize: Int,
+    maxBins: Int,
+    mcSampleSize: Int,
+    numResults: Int): MSEDistributionResult = {
+    val sample = dist.sample(histogramSampleSize).toArray
+    val hist = new StreamingHistogramBuilder(maxBins, 100, 1)
+
+    sample.foreach(hist.update(_))
+
+    val equiDistBins = HistogramUtils.equiDistantBins(sample, maxBins)
+    val streamingBins = HistogramUtils.streamingHistogramBins(hist.build)
+
+    val mses: Array[MSEResult] = (0 until numResults).map { _ =>
+      val mcSample = dist.sample(mcSampleSize)
+      val trueDensities = mcSample.map(dist.pdf).toArray
+      val trueCDFs = mcSample.map(dist.cdf).toArray
+
+      def computeMSE(f: Double => Double, trueValues: Array[Double]): Double =
+        mcSample.map(f).zip(trueValues).map { case (a, b) => math.pow(a - b, 2) }.sum / mcSampleSize
+
+      MSEResult(
+        streamingDensityMSE = computeMSE(HistogramUtils.density(streamingBins), trueDensities),
+        streamingCdfMSE = computeMSE(HistogramUtils.cdf(streamingBins), trueCDFs),
+        equiDistDensityMSE = computeMSE(HistogramUtils.density(equiDistBins), trueDensities),
+        equiDistCdfMSE = computeMSE(HistogramUtils.cdf(equiDistBins), trueCDFs))
+    }.toArray
+
+    val result = MSEDistributionResult(
+      streamingDensityMSE = meanAndVariance(mses.map(_.streamingDensityMSE)),
+      streamingCdfMSE = meanAndVariance(mses.map(_.streamingCdfMSE)),
+      equiDistDensityMSE = meanAndVariance(mses.map(_.equiDistDensityMSE)),
+      equiDistCdfMSE = meanAndVariance(mses.map(_.equiDistCdfMSE)))
+
+    println("-" * 50)
+    println(s"Checking distribution $distributionName " +
+      s"[bins = $maxBins, sample size = $histogramSampleSize, iterations = $numResults]")
+    println("-" * 50)
+    println("Streaming histogram density MSE mean and variance: " +
+      s"${result.streamingDensityMSE.mean}, ${result.streamingDensityMSE.variance}")
+    println("Streaming histogram cdf MSE mean and variance: " +
+      s"${result.streamingCdfMSE.mean}, ${result.streamingCdfMSE.variance}")
+    println("Equidistant histogram density MSE mean and variance: " +
+      s"${result.equiDistDensityMSE.mean}, ${result.equiDistDensityMSE.variance}")
+    println("Equidistant histogram cdf MSE mean and variance: " +
+      s"${result.equiDistCdfMSE.mean}, ${result.equiDistCdfMSE.variance}")
```

Comment:
please add some asserts and change these prints to logs that can be turned on or off